### DISCUSSION TITLE
Best way to show WelcomeView and then show Rootview?

### DISCUSSION THREAD

**Discussion by ms-somanna** [canton7/Stylet#327] [Q&A]
So I am developing an application that is used to create some kind of projects.
I currently have two Views namely `RootView` and `WelcomeView` and one Viewmodel named `RootViewModel`. I would like to couple both the views to the same viewmodel though that is not absolutely necessary. I could also use seperate viewmodels for each view.

What I want to achieve is-
1. When I open the app, it must first show me a Welcome window (WelcomeView) without showing the RootView. In the WelcomeView there is a button. When I click on the button the WelcomeView must close and the RootView must open. To achieve this I could use native technics like setting the StartUp object to WelcomeView and then handle the button command to close the WelcomeView and show the RootView. But is this the way to do it using stylet? Or is there another actual stylet way of doing this?
2. When I click on the Close Button on the title bar of the WelcomeView the app must terminate without showing the RootView. How to achieve this with stylet?

**Comment by canton7** [comment]
The way that Stylet works by default means that it's a lot easier if you have one View per ViewModel. It's possible to change this, but you need to change the way that Stylet maps ViewModels to Views. See the docs page on the ViewManager.

I'm going to rename your RootView/Model to ShellView/Model for the purposes of this answer. 

I'd make your WelcomeView the root view (i.e. your bootstrapper inherits from `Bootstrapper<WelcomeView>`. This means that it's shown on start-up. In the WelcomeViewModel, handle the close button and button click actions. If the close button is clicked, just close the window with `RequestClose()`. If the button is clicked, first show the ShellViewModel using `IWindowManager.ShowWindow`, then close the window as before.

Alternatively, make your bootstrapper inherit from `Bootstrapper<ShellView>`, but override the `Launch` method. Here, do `bool result = Container.GetInstance<IWindowManager>().ShowWindow(new WelcomeViewModel())`, and only call `base.Launch` if `result == true`. Inside `WelcomeViewModel` handle the button and close click actions as before, but call `RequestClose(true)` from the button click, and `RequestClose(false)` from the close button click.
